internal/process/provisioning/kyma_append_modules.go

```diff
@@ -0,0 +1,117 @@
+package provisioning
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/kyma-project/kyma-environment-broker/internal"
+	"github.com/kyma-project/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/kyma-environment-broker/internal/process/steps"
+	"github.com/kyma-project/kyma-environment-broker/internal/storage"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type KymaAppendModules struct {
+	operationManager *process.OperationManager
+	logger           logrus.FieldLogger
+}
+
+var _ process.Step = &KymaAppendModules{}
+
+func (k *KymaAppendModules) Name() string {
+	return "Kyma_Append_Modules"
+}
+
+func NewKymaAppendModules(os storage.Operations) *KymaAppendModules {
+	return &KymaAppendModules{operationManager: process.NewOperationManager(os)}
+}
+
+func (k *KymaAppendModules) Run(operation internal.Operation, logger logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+	k.logger = logger
+	if operation.Type != internal.OperationTypeProvision {
+		k.logger.Infof("%s is supposed to run only for Provisioning, skipping logic.", k.Name())
+		return operation, 0, nil
+	}
+
+	modules := operation.ProvisioningParameters.Parameters.Modules
+	if modules != nil {
+		// First case -> if 'default' is false, then we don't install any modules
+		// Second case -> if 'list' is given and not empty, we append passed modules to Kyma, if 'list' is given and is empty, then we don't install any modules
+		// Any other case will cause usage of default Kyma modules
+		applyList := (modules.Default != nil && !*modules.Default) || (modules.Default == nil && modules.List != nil)
```

Comment:
But is it even possible that modules.Default != nil and modules.List != nil at the same time given that you used anyOf in the schema?
On the other side it is safer/defensive approach.